astropy/io/fits/fitstime.py

```diff
@@ -305,15 +452,28 @@ def fits_to_time(hdr, table):
             time_columns[int(idx)][base] = value
             hcopy.remove(key)
 
-    if len(hcopy) != len(hdr):
-        _verify_global_info(global_info)
-        _convert_global_time(table, global_info)
-        if time_columns:
-            for idx, column_info in time_columns.items():
-                if _verify_column_info(column_info, global_info):
-                    colname = table.colnames[idx - 1]
-                    table[colname] = _convert_time_column(table[colname],
-                                                          column_info)
+    # Verify and get the global time reference frame information
+    _verify_global_info(global_info)
+    _convert_global_time(table, global_info)
+
+    # Columns with column-specific time (coordinate) keywords
+    if time_columns:
+        for idx, column_info in time_columns.items():
+            # Check if the column is time coordinate (not spatial)
+            if _verify_column_info(column_info, global_info):
+                colname = table.colnames[idx - 1]
+                # Convert to Time
+                table[colname] = _convert_time_column(table[colname],
+                                                      column_info)
+
+    # Check for special-cases of time coordinate columns
+    other_columns = [idx for idx in range(1, len(table.columns) + 1) if
+                     idx not in time_columns]
+    for idx in other_columns:
```

Comment:
I think I'd get the names out in the loop and do the skipping more directly using:
```
for idx, colname in enumerate(table.colnames):
    if (idx + 1) not in time_columns:
        column_info = ...
```